### ISSUE TITLE
Seg fault during gameplay

### ISSUE THREAD

**Issue by johndh** [Andrettin/Wyrmgus#75]
```
BUG: RangeTargetFinder::FillBadGood.Compute out of range. size: 22, pos: -4, x: 18, xx: 0, y: -2, yy: 0BUG: RangeTargetFinder::FillBadGood.Compute out of range. size: 22, pos: -3, x: 19, xx: 0, y: -2, yy: 0BUG: RangeTargetFinder::FillBadGood.Compute out of range. size: 22, pos: -2, x: 20, xx: 0, y: -2, yy: 0BUG: RangeTargetFinder::FillBadGood.Compute out of range. size: 22, pos: -1, x: 21, xx: 0, y: -2, yy: 0BUG: RangeTargetFinder::FillBadGood.Compute out of range. size: 22, pos: -4, x: 18, xx: 0, y: -2, yy: 0game ends
Segmentation fault
```
This happened as the enemy was attacking my base in the Moving Into Galicia (Teuton) quest.
Running git version on Linux.

**Comment by Andrettin** [comment]
Thanks for the bug report! Maybe this issue is linked to siege weapons firing; I haven't encountered it so far though. I'll run some more tests.

**Comment by johndh** [comment]
No idea if it's the same cause, but I had another segfault.  The quest was Home of the Boii, and I had no siege weapons.  I don't think there was any combat going on either.

**Comment by KroArtem** [comment]
Sometimes I have `Compute out of range` in the output but it doesn't seem to lead to crashes. Is there any reason to have old tickets related to old versions being open? With some major version you can just close all irrelevant tickets with 'not reproducible' resolution.

**Comment by Andrettin** [comment]
I agree; this particular ticket is still relevant though since I didn't change that part of the code in unit_find.h.
